src/pages/Movies/index.jsx

```diff
@@ -1,72 +1,64 @@
-import { useCallback, useMemo, useState } from "react";
+import { useCallback, useMemo } from "react";
 import { MovieCard } from "../../components/layout/MovieCard";
 import { MoviesAPI } from "../../services/api";
 import { Pagination } from "../../components/layout/Pagination";
 import SearchIcon from "../../assets/svgs/search.svg?react";
-import HeaderFooter from "../../components/layout/HeaderFooter";
 import { useInfiniteQuery } from "@tanstack/react-query";
-import { Loader } from "../../components/ui/Loader";
-import { DisplayError } from "../../components/ui/DisplayError";
-import { Link } from "react-router";
-import { REACT_QUERY_CONFIG } from "../../lib/constants/queryConfig";
+import { Link, useSearchParams } from "react-router";
 import { debounce } from "lodash";
+import { SafeRender } from "../../components/layout/SafeRender";
 
 const Movies = () => {
-  const [searchQuery, setSearchQuery] = useState("");
-  const [debouncedSearch, setDebouncedSearch] = useState("");
+  const [searchParams, setSearchParams] = useSearchParams();
+  const searchQuery = searchParams.get("query") || "";
 
-  const debouncedSetSearch = useCallback(
+  const updateSearch = useCallback(
     debounce((value) => {
-      setDebouncedSearch(value);
+      value.trim() ? setSearchParams({ query: value }) : setSearchParams({});
     }, 500),
-    []
+    [setSearchParams]
   );
 
-  const handleSearchQuery = (e) => {
-    const value = e?.target?.value || "";
-    setSearchQuery(value);
-    debouncedSetSearch(value);
-  };
+  const handleSearchQuery = (e) => updateSearch(e.target.value);
 
   const fetchMovies = async ({ queryKey, pageParam = 1 }) => {
-    const [_key] = queryKey;
-    return debouncedSearch
-      ? await MoviesAPI.getSearchedMovies(debouncedSearch)
+    const [_key, searchTerm] = queryKey;
+    return searchTerm
+      ? await MoviesAPI.getSearchedMovies(searchTerm)
       : await MoviesAPI.getAll(pageParam);
```

Comment:
`await MoviesAPI.getSearchedMovies(searchTerm)`
In this case we are sending only search not page no



```
const getMovies =    searchTerm
      ? ()=> MoviesAPI.getSearchedMovies(searchTerm)
      : ()=> MoviesAPI.getAll(pageParam);
      
return await getMovies()
```